### PR TITLE
Fixed modal mobile view for application. 

### PR DESCRIPTION
**Pull request by cdevent**
## Related Issue or Background Info
Closes https://github.com/CDCgov/prime-simplereport/issues/1574

## Changes Proposed

-fixed mobile modal preview
-fixed buttons positions
-some UI improvements

## Additional Information

- added responsiveness mixins 

## Screenshots / Demos

https://user-images.githubusercontent.com/7571965/118498520-c8ca1e80-b72e-11eb-913b-e99b82ff1a63.mov

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### CODE REVIEW

**Review by maya — CHANGES_REQUESTED** [review]
Can you: 
- Remove the custom mixins and use the [USWDS media query mixins](https://designsystem.digital.gov/utilities/layout-grid/#mixins) instead. You can also use the layout utilities as/if needed
- Write this mobile-first
- Write custom classes using BEM notation

File: frontend/src/scss/settings/_uswds-theme-utilities.scss [hunk 1]
```diff
@@ -36,7 +36,7 @@ $theme-utility-breakpoints: (
   // 240px:
   "card-lg": false,
   // 320px:
-  "mobile": false,
+  "mobile": true,
```
Comment (maya) [inline frontend/src/scss/settings/_uswds-theme-utilities.scss:39]: I think we want to keep this off and the CSS that you're currently using `mobile` on, you don't use a media query. That keeps it mobile-first.

**Comment by maya** [comment]
Doesn't display right on this size:
Screen Shot 2021-05-18 at 11 17 49 AM

Padding is too wide, it should be the same as how it is currently:
Screen Shot 2021-05-18 at 11 20 22 AM

File: frontend/src/scss/settings/_uswds-theme-utilities.scss (hunk 1 above)
Comment (cdevent) [reply frontend/src/scss/settings/_uswds-theme-utilities.scss:39]: Oh, I think now I catch what you mean.

**Comment by cdevent** [comment]
> Doesn't display right on this size:
> Screen Shot 2021-05-18 at 11 17 49 AM
> 
> Padding is too wide, it should be the same as how it is currently:
> Screen Shot 2021-05-18 at 11 20 22 AM

what size is it?

**Comment by maya** [comment]
The size of the modal is `tablet`, so I think you want to change your media queries that are `desktop` to `tablet`. That should take care of some of the other issues.

**Comment by maya** [comment]
this doesn't seem right:
Screen Shot 2021-05-18 at 11 40 43 AM

File: frontend/src/scss/PrimeStyles.scss [hunk 2]
```diff
@@ -789,3 +805,47 @@ $results-dropdown-spacing: #{units(4)} - #{units(2)} - 22px - #{units(4)}; // he
...
+  @include at-media(desktop) {
+    position: fixed;
+    width: unset;
+    right: unset;
+    padding: 0;
+    bottom: units(7);
+  }
+}
+
+.modal__submit--container {
```
Comment (maya) [inline frontend/src/scss/PrimeStyles.scss:830]: unless this is something we want to generalize to other modals, use `aoe-modal` as your "block" in BEM. `container ` is not a modifier of `submit`, it's either another element or a description of submit, so just keep to `submit` or `submit-container`

File: frontend/src/scss/PrimeStyles.scss [hunk 3]
```diff
@@ -789,3 +805,47 @@ $results-dropdown-spacing: #{units(4)} - #{units(2)} - 22px - #{units(4)}; // he
...
+    margin-right: 0 !important;
+  }
+
+  @include at-media(desktop) {
+    box-shadow: unset;
+    margin: unset;
+  }
+}
+
+.section__qr--container {
```
Comment (maya) [inline frontend/src/scss/PrimeStyles.scss:849]: `section` is not a descriptive name, since it's specific to the AOE modal, `aoe-modal__qr` should work

Comment (cdevent) [reply frontend/src/scss/PrimeStyles.scss:849]: Thanks for the review! will fix it on my morning

Comment (cdevent) [reply frontend/src/scss/PrimeStyles.scss:849]: > this doesn't seem right:
> Screen Shot 2021-05-18 at 11 40 43 AM

Can you please describe what the actual size it is? Can't reproduce it on my side

**Comment by cdevent** [comment]
@maya Can you please look on it's again? i've pushed some fixes now. also, please describe on what sizes it's not display normaly? on my side looks ok

**Review by timothybest-usds — APPROVED** [review]

**Comment by cdevent** [comment]
@maya any chance to review it again?

**Comment by maya** [comment]
@cdevent sure, I can re-review later today.

**Comment by maya** [comment]
Had a few things come up today, but should be able to get to this tomorrow.

**Review by maya — CHANGES_REQUESTED** [review]
I added style and naming changes in #1737. Please merge into yours.

The problem I posted earlier is still there. The modal footer obstructs the QR code when the QR code option is selected, making it unusable.
 
 This is the issue in this PR:
Screen Shot 2021-05-27 at 5 37 14 PM

This is what currently working in `main` and needs to return to (not sticky):
qXKtj325Kv

Another issue with the texting option, it's too tall on large screen sizes.

This is what you have:
Screen Shot 2021-05-27 at 5 49 11 PM

This is what it should look like (not sticky and current on `main`):
Screen Shot 2021-05-27 at 5 44 09 PM

**Comment by cdevent** [comment]
@maya IMO, the bottom panel should remain on screen in any way, to allow click on continue button without scrolling to it. I propose to scroll in view QR when it selected. will push some changes to look how it should be IMO. 
btw thanks that you mentioned modal size, yes, it should be full-screen size only on a mobile. I'll fix that

**Comment by cdevent** [comment]
@maya Now I've figured out why it was broken on your side) are you using Mozilla? because on Chrome and Safari it looks good, but on Mozilla, it's really weird. Working on fixes on moz side too =)

**Comment by cdevent** [comment]
@maya I've left a sticky footer for mobile-only and also fix the view on Mozilla. now it should look OK. But I want to propose to add in a separate PR, auto-focus on QR(scroll it into view when it's mounted). 

https://user-images.githubusercontent.com/7571965/119950248-587b9280-bfa3-11eb-9604-d72fe9be2093.mov

**Comment by cdevent** [comment]
I also mentioned that on Mozilla margins for labels and legends are too high, so I've pushed some fix for moz-document only too. 

Screenshot 2021-05-28 at 12 03 38 PM

**Comment by maya** [comment]
Looking better! I use Chrome, not Firefox but your changes fixed it for me.

A few style changes I pushed up:

- Update the footer on QR and text option so the border extends full width on large screens
- Removed duplicate 0 top margins on the moz `.usa-label` (we already do that globally) -- the difference in FF is bc of how FF doesn't collapse fieldset/legend margins and webkit browsers do
- Wrapped the new date picker in a `.usa-form-group` so it has the proper top margins in the AoE form
- Added bottom padding to the bottom of the AoE form on large screens b/c it was too close to the bottom

**Comment by maya** [comment]
I like your idea of scrolling the QR code into view after selecting that option.

**Comment by cdevent** [comment]
> I like your idea of scrolling the QR code into view after selecting that option.

Let's add it in a separate PR to finish up with this one.

**Review by maya — APPROVED** [review]

**Comment by maya** [comment]
PR approved, great work 🚀

**Pull request merged**